### PR TITLE
Project og title added in ssr

### LINKED ISSUE #2177: Improve OG title for linked p5.js sketches
### **Problem description** 
When sharing a p5.js sketch from the web editor on social media or other websites, the OG title of the resulting webpage only says "p5.js Web Editor". This makes it difficult to distinguish between different linked sketches or know what the sketch is about.

### **Proposed solution** 
Update the OG title to include the title of the sketch and the name of the author, followed by "p5.js Web Editor". For example, the OG title could be "My Sketch by Username - p5.js Web Editor". This would improve the discoverability and sharing of p5.js sketches and help to promote the creators of these sketches.

### How would this new feature help [increase access](https://github.com/processing/p5.js/blob/main/contributor_docs/access.md) to the p5.js Editor?

The new feature would help increase access to the p5.js Editor by making it easier for users to share and promote their p5.js sketches on social media and other websites. By improving the discoverability and sharing of p5.js sketches, more people will be exposed to the capabilities of p5.js, and may be inspired to try it out for themselves.

### **Additional information** 
Screenshot from the Processing Foundation forum showing the current behavior is attached below

image

### PR DESCRIPTION
**Pull request by dewanshDT**
Fixes #2177 

Changes:
modified the ssr to show the Open Graph title. please check the file changes for details.

I have verified that this pull request:

* [x] has no linting errors (`npm run lint`)
* [x] has no test errors (`npm run test`)
* [x] is from a uniquely-named feature branch and is up to date with the  `develop` branch.
* [x] is descriptively named and links to an issue number, i.e. `Fixes #123`

### CODE REVIEW

File: server/views/index.js [hunk 1]
```diff
@@ -23,25 +28,132 @@ export function renderIndex() {
...
+        };
         window.process.env.CLIENT = true;
-        window.process.env.LOGIN_ENABLED = ${process.env.LOGIN_ENABLED === 'false' ? false : true};
-        window.process.env.EXAMPLES_ENABLED = ${process.env.EXAMPLES_ENABLED === 'false' ? false : true};
-        window.process.env.UI_ACCESS_TOKEN_ENABLED = ${process.env.UI_ACCESS_TOKEN_ENABLED === 'false' ? false : true};
-        window.process.env.UI_COLLECTIONS_ENABLED = ${process.env.UI_COLLECTIONS_ENABLED === 'false' ? false : true};
-        window.process.env.UPLOAD_LIMIT = ${process.env.UPLOAD_LIMIT ? `${process.env.UPLOAD_LIMIT}` : undefined};
-        window.process.env.MOBILE_ENABLED = ${process.env.MOBILE_ENABLED ? `${process.env.MOBILE_ENABLED}` : undefined};
-        window.process.env.TRANSLATIONS_ENABLED = ${process.env.TRANSLATIONS_ENABLED === 'true' ? true : false}; 
+        window.process.env.LOGIN_ENABLED = ${
```
Comment (dewanshDT) [inline server/views/index.js:41]: the file was not formatted before for  some reason

File: server/views/index.js [hunk 2]
```diff
@@ -23,25 +28,132 @@ export function renderIndex() {
...
+        process.env.NODE_ENV === 'production'
+          ? `${assetsManifest['/app.js']}`
+          : '/app.js'
+      }'></script>
+    </body>
+  </html>
+  `;
+}
+
+export function renderProjectIndex(projectName, username) {
```
Comment (dewanshDT) [inline server/views/index.js:81]: added this function for rendering the user's Project index file

File: server/views/index.js [hunk 3]
```diff
@@ -23,25 +28,132 @@ export function renderIndex() {
...
+      <meta name="viewport" content="width=device-width, initial-scale=1.0">
+      <meta name="keywords" content="p5.js, p5.js web editor, web editor, processing, code editor" />
+      <meta name="description" content="A web editor for p5.js, a JavaScript library with the goal of making coding accessible to artists, designers, educators, and beginners." />
+      <meta property="og:title" content="${`${projectName} by ${username} -`} p5.js Web Editor" />
+      <title>${`${projectName} by ${username} -`} p5.js Web Editor</title>
+      ${
+        process.env.NODE_ENV === 'production'
+          ? `<link rel='stylesheet' href='${assetsManifest['/app.css']}' />`
+          : ''
+      }
```
Comment (dewanshDT) [inline server/views/index.js:98]: modified title and also added og:title

File: server/views/index.js [hunk 4]
```diff
@@ -23,25 +28,132 @@ export function renderIndex() {
...
+  return `
+  <!DOCTYPE html>
+  <html lang="en">
+    <head>
+      <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+      <meta name="viewport" content="width=device-width, initial-scale=1.0">
+      <meta name="keywords" content="p5.js, p5.js web editor, web editor, processing, code editor" />
+      <meta name="description" content="A web editor for p5.js, a JavaScript library with the goal of making coding accessible to artists, designers, educators, and beginners." />
+      <meta property="og:title" content="${`${projectName} by ${username} -`} p5.js Web Editor" />
+      <title>${`${projectName} by ${username} -`} p5.js Web Editor</title>
```
Comment (dewanshDT) [inline server/views/index.js:93]: modified title and also added og:title

**Comment by raclim** [comment]
Thanks for your work on this and sorry for getting to this so late! 

After looking through it, I think some of the code in the `server/views/index.js` file feels a little redundant and could probably be more precise. For example, I think an index file title containing a user's project could potentially be determined by a conditional operator instead of making a new `renderProjectIndex()` function. 

I'm going to close this one for now, but please feel free to revisit this and reopen it or create a new one in the future, thanks again!

**Pull request closed without merging**